homeassistant/components/homekit/type_cameras.py

```diff
@@ -240,6 +263,26 @@ def _async_update_motion_state(self, new_state):
             detected,
         )
 
+    @callback
+    def _async_update_doorbell_state_event(self, event):
+        """Handle state change event listener callback."""
+        self._async_update_doorbell_state(event.data.get("new_state"))
+
+    @callback
+    def _async_update_doorbell_state(self, new_state):
+        """Handle link doorbell sensor state change to update HomeKit value."""
+        if not new_state:
+            return
+
+        if new_state.state == STATE_ON:
+            self._char_doorbell_detected.set_value(DOORBELL_SINGLE_PRESS)
```

Comment:
```suggestion
            self._char_doorbell_detected.set_value(DOORBELL_SINGLE_PRESS)
            self._char_doorbell_detected.set_value(None, should_notify=False)
```

Reply:
So this actually broke the tests, due to None being an invalid value. The only allowed values are 0,1,2. Was this the line that fixed the notifications for you, or was is setting the primary service?

Reply:
The spec is very clear about this returning `null` and only sending events.

If we set valid_values when we create the char, does it pass?

Reply:
```
2020-07-31 18:56:32 DEBUG (Thread-80) [pyhap.hap_server] finish_request: '<' not supported between instances of 'int' and 'NoneType'
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/pyhap/hap_server.py", line 943, in finish_request
    self.RequestHandlerClass(request, client_address,
  File "/usr/local/lib/python3.8/site-packages/pyhap/hap_server.py", line 164, in __init__
    super(HAPServerHandler, self).__init__(sock, client_addr, server)
  File "/usr/local/lib/python3.8/socketserver.py", line 720, in __init__
    self.handle()
  File "/usr/local/lib/python3.8/http/server.py", line 429, in handle
    self.handle_one_request()
  File "/usr/local/lib/python3.8/http/server.py", line 415, in handle_one_request
    method()
  File "/usr/local/lib/python3.8/site-packages/pyhap/hap_server.py", line 270, in dispatch
    getattr(self, self.HANDLERS[self.command][path])()
  File "/usr/local/lib/python3.8/site-packages/pyhap/hap_server.py", line 555, in handle_accessories
    hap_rep = self.accessory_handler.get_accessories()
  File "/usr/local/lib/python3.8/site-packages/pyhap/accessory_driver.py", line 632, in get_accessories
    hap_rep = self.accessory.to_HAP()
  File "/usr/local/lib/python3.8/site-packages/pyhap/accessory.py", line 366, in to_HAP
    return [acc.to_HAP() for acc in (super(), *self.accessories.values())]
  File "/usr/local/lib/python3.8/site-packages/pyhap/accessory.py", line 366, in <listcomp>
    return [acc.to_HAP() for acc in (super(), *self.accessories.values())]
  File "/usr/local/lib/python3.8/site-packages/pyhap/accessory.py", line 235, in to_HAP
    HAP_REPR_SERVICES: [s.to_HAP() for s in self.services],
  File "/usr/local/lib/python3.8/site-packages/pyhap/accessory.py", line 235, in <listcomp>
    HAP_REPR_SERVICES: [s.to_HAP() for s in self.services],
  File "/usr/local/lib/python3.8/site-packages/pyhap/service.py", line 90, in to_HAP
    HAP_REPR_CHARS: [c.to_HAP() for c in self.characteristics],
  File "/usr/local/lib/python3.8/site-packages/pyhap/service.py", line 90, in <listcomp>
    HAP_REPR_CHARS: [c.to_HAP() for c in self.characteristics],
  File "/usr/local/lib/python3.8/site-packages/pyhap/characteristic.py", line 250, in to_HAP
    sorted(self.properties[PROP_VALID_VALUES].values())
TypeError: '<' not supported between instances of 'int' and 'NoneType'
```

Sadly now seeing this.

Reply:
Test passes, but runtime errors occur: 

```Exception happened during processing of request from ('172.16.1.31', 49413)
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.4/Frameworks/Python.framework/Versions/3.8/lib/python3.8/socketserver.py", line 650, in process_request_thread
    self.finish_request(request, client_address)
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/hap_server.py", line 943, in finish_request
    self.RequestHandlerClass(request, client_address,
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/hap_server.py", line 164, in __init__
    super(HAPServerHandler, self).__init__(sock, client_addr, server)
  File "/usr/local/Cellar/python@3.8/3.8.4/Frameworks/Python.framework/Versions/3.8/lib/python3.8/socketserver.py", line 720, in __init__
    self.handle()
  File "/usr/local/Cellar/python@3.8/3.8.4/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/server.py", line 429, in handle
    self.handle_one_request()
  File "/usr/local/Cellar/python@3.8/3.8.4/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/server.py", line 415, in handle_one_request
    method()
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/hap_server.py", line 270, in dispatch
    getattr(self, self.HANDLERS[self.command][path])()
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/hap_server.py", line 555, in handle_accessories
    hap_rep = self.accessory_handler.get_accessories()
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/accessory_driver.py", line 632, in get_accessories
    hap_rep = self.accessory.to_HAP()
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/accessory.py", line 366, in to_HAP
    return [acc.to_HAP() for acc in (super(), *self.accessories.values())]
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/accessory.py", line 366, in <listcomp>
    return [acc.to_HAP() for acc in (super(), *self.accessories.values())]
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/accessory.py", line 235, in to_HAP
    HAP_REPR_SERVICES: [s.to_HAP() for s in self.services],
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/accessory.py", line 235, in <listcomp>
    HAP_REPR_SERVICES: [s.to_HAP() for s in self.services],
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/service.py", line 90, in to_HAP
    HAP_REPR_CHARS: [c.to_HAP() for c in self.characteristics],
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/service.py", line 90, in <listcomp>
    HAP_REPR_CHARS: [c.to_HAP() for c in self.characteristics],
  File "/Users/austin/Dev/GitHub/home-assistant-core/venv/lib/python3.8/site-packages/pyhap/characteristic.py", line 250, in to_HAP
    sorted(self.properties[PROP_VALID_VALUES].values())
TypeError: '<' not supported between instances of 'int' and 'NoneType'```